Framework/DataHandling/src/SaveAscii2.cpp

```diff
@@ -603,4 +607,75 @@ void SaveAscii2::writeFileHeader(const std::vector<std::string> &logList, std::o
   outputFile << '\n';
 }
 
+void SaveAscii2::write1DHistoCut(const IMDHistoWorkspace_const_sptr &mdws, const std::string &filename,
+                                 bool appendToFile, bool writeHeader, int prec, bool scientific,
+                                 const std::string &comment) {
+  int count = 0;
+  size_t dimIndex = -1;
+  for (size_t i = 0; i < mdws->getNumDims(); i++) {
+    if (mdws->getDimension(i)->getNBins() > 1) {
```

Comment:
I just found that an MD workspace has a method `getNonIntegratedDimensions()` that returns a vector - so we can check the length of that vector is 1 and also then we don't need to do this loop to get the index of the non-integrated dimension!